ORBIT/phases/design/SemiTaut_mooring_system_design.py

```diff
@@ -0,0 +1,167 @@
+
+    def __init__(self, config, **kwargs):
+        """
+        Creates an instance of MooringSystemDesign.
+
+        Parameters
+        ----------
+        config : dict
+        """
+
+        config = self.initialize_library(config, **kwargs)
+        self.config = self.validate_config(config)
+        self.num_turbines = self.config["plant"]["num_turbines"]
+
+        self._design = self.config.get("mooring_system_design", {})
+        self.num_lines = self._design.get("num_lines", 4)
+        self.anchor_type = self._design.get("anchor_type", "Drag Embedment")
+
+        self._outputs = {}
+
+    def run(self):
+        """
+        Main run function.
+        """
+
+        self.calculate_line_length_mass()
+        self.determine_mooring_line_cost()
+        self.calculate_anchor_mass_cost()
+
+        self._outputs["mooring_system"] = {**self.design_result}
+
+    def calculate_line_length_mass(self):
+        """
+        Returns the mooring line length and mass.
+        """
+
+        depth = self.config["site"]["depth"]
+
+        # Input hybrid mooring system design from Cooperman et al. (2022), https://www.nrel.gov/docs/fy22osti/82341.pdf 'Assessment of Offshore Wind Energy Leasing Areas for Humboldt and Moorow Bay Wind Energy Areas, California
+        depths = np.array([500, 750, 1000, 1250, 1500])
+        rope_lengths = np.array([478.41, 830.34, 1229.98, 1183.93, 1079.62])
+        rope_diameters = np.array([0.2, 0.2, 0.2, 0.2, 0.2]) # you need the diameter for the cost data
+        chain_lengths = np.array([917.11, 800.36, 609.07, 896.42, 1280.57])
+        chain_diameters = np.array([0.13, 0.17, 0.22, 0.22, 0.22])
+        
+        # Interpolate
+        finterp_rope = interp1d(depths, rope_lengths)
+        finterp_chain = interp1d(depths, chain_lengths)
+        finterp_rope_diam = interp1d(depths, rope_diameters)
+        finterp_chain_diam = interp1d(depths, chain_diameters)
+        
+        # Rope and chain length at project depth
+        self.chain_length = finterp_chain(depth)
+        self.rope_length = finterp_rope(depth)
+        # Rope and chain diameter at project depth
+        self.rope_diameter = finterp_rope_diam(depth)
+        self.chain_diameter = finterp_chain_diam(depth)
+        
+        self.line_length = self.rope_length + self.chain_length
+        
+        chain_kg_per_m = 19900 * (self.chain_diameter**2) # 19,900 kg/m^2 (diameter)/m (length)
+        rope_kg_per_m = 797.8 * (self.rope_diameter**2) # 797.8 kg/ m^2 (diameter) / m (length) 
+        self.line_mass = (self.chain_length * chain_kg_per_m) + (self.rope_length * rope_kg_per_m) # kg
+        #print('total hybrid line mass is ' + str(self.line_mass) + 'kg')
+        # convert kg to metric tonnes
+        self.line_mass = self.line_mass/1e3
+        
+    def calculate_anchor_mass_cost(self):
+        """
+        Returns the mass and cost of anchors.
+
+        TODO: Anchor masses are rough estimates based on initial literature
```

Comment:
Yes. This is still relevant. I included the TODO in `mooring_system_design.py` just to keep track of it. Solving it would involve further research on anchor types.